From 3759f24379c98b984a56b52c7d23a2dfbf36efcf Mon Sep 17 00:00:00 2001
Message-Id: <3759f24379c98b984a56b52c7d23a2dfbf36efcf.1372337066.git.stefano.lattarini <at> gmail.com>
From: Stefano Lattarini <stefano.lattarini <at> gmail.com>
Date: Thu, 27 Jun 2013 14:43:12 +0200
Subject: [PATCH] tests: avoid a spurious failure on MacOS X 10.6.8

Fixes automake bug#14706.

* t/depcomp2.sh: Strip, from the redirected ./configure stderr, the
possible error message "rm: conftest.dSYM: is a directory", generated
by cleanup code that doesn't cater to the existence of *.dSYM
directories sometimes created by the compiler on MacOS X.  This
"massaging" of ./configure stderr is legitimate, since the spurious
error message is due not to automake-related code, but to a know
buglet/limitation of either Autoconf or Mac OS X bundles gcc:
<http://lists.gnu.org/archive/html/bug-autoconf/2007-11/msg00017.html>
Actually, from that link it appears that the original Autoconf issue
had been fixed, but it must have been re-introduced in the meantime :-(

Signed-off-by: Stefano Lattarini <stefano.lattarini <at> gmail.com>
---
 t/depcomp2.sh | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/t/depcomp2.sh b/t/depcomp2.sh
index 2182164..3eba12d 100644
--- a/t/depcomp2.sh
+++ b/t/depcomp2.sh
@@ -45,6 +45,9 @@ $AUTOMAKE --add-missing
 $AUTOCONF
 ./configure 2>stderr || { cat stderr >&2; exit 1; }
 cat stderr >&2
-test ! -s stderr
+# Ignore warning messages sometimes seen on Mac OS X; they are
+# not automake's fault anyway, but either autoconf's or Mac's.
+sed '/rm:.* conftest\.dSYM/d' stderr >stderr2
+test -s stderr2 && { cat stderr2; exit 1; }

 :
-- 
1.8.3.1.448.gfb7dfaa

From 4d7dcafc0f419378cd80e46f9390950c6fbaffa0 Mon Sep 17 00:00:00 2001
Message-Id: <4d7dcafc0f419378cd80e46f9390950c6fbaffa0.1374421729.git.stefano.lattarini@gmail.com>
From: Stefano Lattarini <stefano.lattarini@gmail.com>
Date: Sun, 21 Jul 2013 13:46:48 +0100
Subject: [PATCH] test: avoid false positives in 'cc-no-c-o' script

Fixes automake bug#14991.

* t/ax/cc-no-c-o.in: Be more careful in determining whether both the
'-c' and '-o' options have been passed on the command line to the
compiler.  In particular, do not spuriously complain in the face of
options like '-compatibility_version' or '-current_version' (seen on
Mac OS X 10.7).
* THANKS: Update.

Signed-off-by: Stefano Lattarini <stefano.lattarini@gmail.com>
---
 THANKS            |  1 +
 t/ax/cc-no-c-o.in | 20 ++++++++++++++++----
 2 files changed, 17 insertions(+), 4 deletions(-)

diff --git a/THANKS b/THANKS
index b708943..1482da2 100644
--- a/THANKS
+++ b/THANKS
@@ -145,6 +145,7 @@ Gwenole Beauchesne              gbeauchesne@mandrakesoft.com
 H.J. Lu                         hjl@lucon.org
 H.Merijn Brand                  h.m.brand@hccnet.nl
 Hans Ulrich Niedermann          hun@n-dimensional.de
+Hanspeter Niederstrasser        fink@snaggledworks.com
 Harald Dunkel                   harald@CoWare.com
 Harlan Stenn                    Harlan.Stenn@pfcs.com
 He Li                           tippa000@yahoo.com
diff --git a/t/ax/cc-no-c-o.in b/t/ax/cc-no-c-o.in
index c18f9b9..bbc9ec9 100644
--- a/t/ax/cc-no-c-o.in
+++ b/t/ax/cc-no-c-o.in
@@ -19,11 +19,23 @@

 am_CC=${AM_TESTSUITE_GNU_CC-'@GNU_CC@'}

-case " $* " in
- *\ -c*\ -o* | *\ -o*\ -c*)
+seen_c=false
+seen_o=false
+
+for arg
+do
+  case $arg in
+    -c)
+      seen_c=true;;
+    # It is acceptable not to leave a space between the '-o' option
+    # and its argument, so we have to cater for that.
+    -o|-o*)
+      seen_o=true;;
+  esac
+  if $seen_c && $seen_o; then
     echo "$0: both '-o' and '-c' seen on the command line" >&2
     exit 2
-    ;;
-esac
+  fi
+done

 exec $am_CC "$@"
-- 1.8.3.1.605.g85318f5 
